node/src/test_utils/neighborhood_test_utils.rs

```diff
@@ -274,3 +279,19 @@ impl From<&NodeRecord> for AccessibleGossipRecord {
         }
     }
 }
+
+pub fn make_cpm_recipient() -> (Recipient<ConnectionProgressMessage>, Arc<Mutex<Recording>>) {
+    let (recorder, _, recording_arc) = make_recorder();
+    let addr = recorder.start();
+    let recipient = addr.recipient();
+
+    (recipient, recording_arc)
+}
+
+pub fn make_node_to_ui_recipient() -> (Recipient<NodeToUiMessage>, Arc<Mutex<Recording>>) {
+    let (recorder, _, recording_arc) = make_recorder();
+    let addr = recorder.start();
+    let recipient = addr.recipient();
+
+    (recipient, recording_arc)
+}
```

Comment:
You could write:

```
fn make_recipient_and_recording<M:Message>()->(Recipient<M>,Arc<Mutex<Recording>)
{
    let (recorder, _, recording_arc) = make_recorder();
    let addr = recorder.start();
    let recipient = addr.recipient();

    (recipient, recording_arc)
}
```
Then you could stick it in two tiny functions like:
```
pub fn make_cpm_recipient() -> (Recipient<ConnectionProgressMessage>, Arc<Mutex<Recording>>) {
   make_recipient_and_recording()
}
```
...and the other, similarly...

Reply:
This is what the Rust compiler is accepting in the end:

```rust
pub fn make_recipient_and_recording_arc<M: 'static>() -> (Recipient<M>, Arc<Mutex<Recording>>)
where
    M: Message + std::marker::Send,
    <M as actix::Message>::Result: std::marker::Send,
    recorder::Recorder: actix::Handler<M>,
{
    let (recorder, _, recording_arc) = make_recorder();
    let addr = recorder.start();
    let recipient = addr.recipient::<M>();

    (recipient, recording_arc)
}

pub fn make_cpm_recipient() -> (Recipient<ConnectionProgressMessage>, Arc<Mutex<Recording>>) {
    make_recipient_and_recording_arc()
}

pub fn make_node_to_ui_recipient() -> (Recipient<NodeToUiMessage>, Arc<Mutex<Recording>>) {
    make_recipient_and_recording_arc()
}
```

Reply:
In general I don't mind to have those boundaries specified if they are needed. You can make the boundaries briefer like this:
 M: Message + Send,
 <M as actix::Message>::Result: Send,
'static

To be honest I don't understand this kind of boundary so please inspect if it is necessary...by temporarily removing it.    
Recorder: actix::Handler<M>,